### PR TITLE
[Keyboard] Add SP60

### PR DESCRIPTION
**Pull request by jiaxin96**
## Description
Add SP60 Rev1.0 and Rev2.0 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yandrstudio/sp60/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "sp60",
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/info.json:2]:
```suggestion
    "keyboard_name": "sp60",
    "manufacturer": "QMK",
    "usb": {
        "vid": "0xFEED"
    },
```

File: keyboards/yandrstudio/sp60/rev1/board.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include_next "board.h"
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/rev1/board.h:6]:
```suggestion
#include_next <board.h>
```

File: keyboards/yandrstudio/sp60/rev1/config.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+#include "config_common.h"
+
+#ifdef RGB_MATRIX_ENABLE
+
+#    define RGB_DI_PIN A7
+#    define RGB_MATRIX_LED_COUNT 65
+
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200
+#    define RGBLIGHT_VAL_STEP 5
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/rev1/config.h:14]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev1/config.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED
+#    define RGB_MATRIX_CENTER { 112, 32 }
+#    define RGB_MATRIX_KEYPRESSES
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#    define RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/rev1/config.h:19]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev2/board.h [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include_next "board.h"
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/rev2/board.h:6]:
```suggestion
#include_next <board.h>
```

File: keyboards/yandrstudio/sp60/rev2/config.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+#ifdef RGBLIGHT_ENABLE
+
+#    define RGB_DI_PIN A9
+#    define RGBLED_NUM 31
+#    define DRIVER_LED_TOTAL RGBLED_NUM
+#    define RGBLIGHT_LIMIT_VAL 180
+#    define RGBLIGHT_SLEEP
+
+
+#    define RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/yandrstudio/sp60/rev2/config.h:17]:
```suggestion
#    define RGBLIGHT_LIMIT_VAL 180
#    define RGBLIGHT_SLEEP
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/sp60/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+#if defined(KEYBOARD_yandrstudio_sp60_rev1)
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/keymaps/default/keymap.c:7]: This is ... gross.

Should move the keymap folder in the rev1 folder, and then make a copy of it in the rev2 folder, and remove the preprocessor stuff.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/sp60/rev1/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev1/keymaps/default/keymap.c:5]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev1/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev1/keymaps/via/keymap.c:5]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev2/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "rev2.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev2/keymaps/default/keymap.c:5]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev2/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "rev2.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev2/keymaps/via/keymap.c:5]:
```suggestion
```

File: keyboards/yandrstudio/sp60/sp60.h [hunk 12]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/sp60.h:6]:
```suggestion
#include "quantum.h"
#if defined(KEYBOARD_yandrstudio_sp60_rev1)
#    include "rev1.h"
#if defined(KEYBOARD_yandrstudio_sp60_rev2)
#    include "rev2.h"
#endif
```

File: keyboards/yandrstudio/sp60/rev1/rev1.h [hunk 13]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev1/rev1.h:6]:
```suggestion
#include "sp60.h"
```

File: keyboards/yandrstudio/sp60/rev2/rev2.h [hunk 14]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev2/rev2.h:6]:
```suggestion
#include "sp60.h"
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/yandrstudio/sp60/rev1/config.h [hunk 15]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev1/config.h:6]:
```suggestion
```

File: keyboards/yandrstudio/sp60/rev2/config.h [hunk 16]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/yandrstudio/sp60/rev2/config.h:6]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/sp60/sp60.c [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+                host_consumer_send(0);
+            }
+            return false;
+        case KC_LPAD:
+            if (record->event.pressed) {
+                host_consumer_send(0x2A0);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
```
Comment (fauxpark) [inline keyboards/yandrstudio/sp60/sp60.c:34]: These keycodes are now in core.

File: keyboards/yandrstudio/sp60/rev2/rules.mk [hunk 18]
```diff
@@ -0,0 +1,2 @@
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+WS2812_DRIVER = pwm         # WS2812 RGB Driver
```
Comment (fauxpark) [inline keyboards/yandrstudio/sp60/rev2/rules.mk:2]: Should go in info.json as `ws2812.driver`.

File: keyboards/yandrstudio/sp60/rev2/rev2.h [hunk 19]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "sp60.h"
+
+#define LAYOUT_60_tsangan_split_rshift( \
```
Comment (fauxpark) [inline keyboards/yandrstudio/sp60/rev2/rev2.h:8]: Layout macros should be moved to into.json by adding `"matrix": [row,col]` to each key, and this file can be deleted.

File: keyboards/yandrstudio/sp60/rev2/rev2.c [hunk 20]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rev2.h"
```
Comment (fauxpark) [inline keyboards/yandrstudio/sp60/rev2/rev2.c:4]:
```suggestion
#include "quantum.h"
```
After deleting the .h file.

File: keyboards/yandrstudio/sp60/rev1/rules.mk [hunk 21]
```diff
@@ -0,0 +1,3 @@
+RGB_MATRIX_ENABLE = yes     # Enable keyboard RGB Matrix
+RGB_MATRIX_DRIVER = WS2812  # RGB matrix driver support
+WS2812_DRIVER = pwm         # WS2812 RGB Driver
```
Comment (fauxpark) [inline keyboards/yandrstudio/sp60/rev1/rules.mk:3]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**